clang/test/CodeGen/target-data.c

```diff
@@ -270,5 +270,5 @@
 // VE: target datalayout = "e-m:e-i64:64-n32:64-S128-v64:64:64-v128:64:64-v256:64:64-v512:64:64-v1024:64:64-v2048:64:64-v4096:64:64-v8192:64:64-v16384:64:64"
 
 // RUN: %clang_cc1 -triple spirv64-amd -o - -emit-llvm %s | \
-// RUN: FileCheck %s -check-prefix=SPIR64
-// AMDGPUSPIRV64: target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-G1-P4-A0"
+// RUN: FileCheck %s -check-prefix=AMDGPUSPIRV64
+// AMDGPUSPIRV64: target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-G1"
```

Comment:
This change is incorrect, the DataLayout differs and we do encode both the Program AS and the alloca AS, please see <https://github.com/llvm/llvm-project/blob/f14ad74478494cbd9d23af841d6f6b2808afda71/clang/lib/Basic/Targets/SPIR.h#L384>, and please don't do this.

Reply:
Hmm, if so, then there might be some bug ? because apparently the test pass with `SPIR64` datalayout, no `P4-A0` are seen.

Reply:
Do you mind take over to fix the bug (also the prefix ?) . I can cancel this PR if you like. Thanks.

Reply:
Sure, I can do that; the issue with the test is that we pick whether it's just plain SPIRV or AMDGCN flavoured SPIRV based on the OS, not on the vendor, please see <https://github.com/llvm/llvm-project/blob/d3ca779d7264bf0751eb4db632c14d8bc05608ca/clang/lib/Basic/Targets.cpp#L677>, so the fix here is to use the `spirv64-amd-amdhsa` triple for the AMDGPUSPIRV64 case. I can do that in this PR or spin-up a new one, whichever you prefer.

Reply:
Updated. Thanks.